src/pkgcheck/checks/metadata.py

```diff
@@ -586,6 +587,70 @@ def feed(self, pkgs):
             yield MissingLocalUseDesc(sorted(missing_desc), pkg=pkg)
 
 
+class UseFlagWithoutDeps(results.VersionResult, results.Warning):
+    """Special Use flag without effect on dependencies and provide little
+    utility.
+
+    Various USE flags, such as "ipv6", should be always turned on or off, and
+    their existence is questionable, in cases were it doesn't introduce new
+    dependencies. Other USE flags, such as "bash-completion", without any new
+    dependencies, are probable violators of small files QA policy [#]_.
+
+    In cases where this USE flag is appropriate, you can silence this warning
+    by adding a description to this USE flag in ``metadata.xml`` file and thus
+    making it a local USE flag instead of global one.
+
+    .. [#] https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0301
+    """
+
+    def __init__(self, flags, **kwargs):
+        super().__init__(**kwargs)
+        self.flags = tuple(flags)
+
+    @property
+    def desc(self):
+        s = pluralism(self.flags)
+        flags = ', '.join(self.flags)
+        return f'special small-files USE flag{s} without effect on dependencies: [ {flags} ]'
+
+
+class UseFlagsWithoutEffectsCheck(GentooRepoCheck):
+    """Check for USE flags without effects."""
```

Comment:
This seems a bit thin, but I guess we have good descriptions in results, and we shouldn't duplicate it.

Reply:
This should get improved when I add the check for zombies (the other issue)